### PR TITLE
Print error message when there is nothing to uninstall

### LINKED ISSUE #5994: linkerd uninstall should have a web link for guidance
### What problem are you trying to solve?

We're trying to remove linkerd2 (or rather, we tried to remove it, and then cert-manager and some other things ended up broken).

Specifically we:
1. disabled linkerd in all namespaces (we knew about, we missed `linkerd-viz`) except `linkerd`
2. restarted as many of our deployments/replicasets as we could (we missed two)
3. deleted the linkerd helm chart and namespace
4. tried upgrading cert-manager
5. discovered that it was very unhappy:
   ```
   Error creating: Internal error occurred: failed calling webhook "linkerd-proxy-injector.linkerd.io": Post https://linkerd-proxy- 
   injector.linkerd.svc:443/?timeout=30s: service "linkerd-proxy-injector" not found
   ```
6. were told to run: `linkerd uninstall |kubectl delete -f -`
   - this fails because the output from `linkerd uninstall` was:
		```
		Please uninstall the following extensions before uninstalling the control-plane:
			* linkerd-viz
		Please uninject the following pods before uninstalling the control-plane:
			* oauth2-proxy-679d98674-dv9f2
		```
7. Was told to run `linkerd uninstall --force`
   - it's unclear if this is the right sequence (do this before or after trying to do something to linkerd-viz?)
8. Foolishly tried running `linkerd uninstall | kubectl apply -f -` -- this is a user error, but highlighting that the verb is `delete` would be useful in a web page.
9. Ran `linkerd uninstall`
	```
	Please uninstall the following extensions before uninstalling the control-plane:
		* linkerd-viz
	```
10. Still have absolutely no idea how to uninstall this
11. Tried running `linkerd viz uninstall`
12. This yielded nothing
13. `kubectl api-resources|grep linkerd` still shows stuff:
	```
	cronjobs                          cj                 tap.linkerd.io/v1alpha1                true         Tap
	daemonsets                        ds                 tap.linkerd.io/v1alpha1                true         Tap
	deployments                       deploy             tap.linkerd.io/v1alpha1                true         Tap
	jobs                                                 tap.linkerd.io/v1alpha1                true         Tap
	namespaces                        ns                 tap.linkerd.io/v1alpha1                false        Tap
	pods                              po                 tap.linkerd.io/v1alpha1                true         Tap
	replicasets                       rs                 tap.linkerd.io/v1alpha1                true         Tap
	replicationcontrollers            rc                 tap.linkerd.io/v1alpha1                true         Tap
	services                          svc                tap.linkerd.io/v1alpha1                true         Tap
	statefulsets                      sts                tap.linkerd.io/v1alpha1                true         Tap
	```

### How should the problem be solved?

What do you want to happen? Add any considered drawbacks.

`linkerd uninstall` should include a link to steps. It could be one page w/ lots of cases, or it could be individual links/anchors for different cases.

### Any alternatives you've considered?

Is there another way to solve this problem that isn't as good a solution?
`linkerd uninstall` could include all the text and case handling internally.

### How would users interact with this feature?

If you can, explain how users will be able to use this. Maybe some sample CLI
output?

### PR DESCRIPTION
**Pull request by adleong**
Fixes #5994 

When running an uninstall command for Linkerd or a Linkerd extension and there are no resources to delete the command silently exits and prints no output.  This can be confusing.

We update the uninstall commands to print an error message to stderr if there are no resources to delete.  Since this message is printed to stderr instead of stdout, it does not interfere with piping the output to kubectl.

```console
> linkerd viz uninstall | kubectl delete -f -
Error: no resources found to uninstall
Usage:
  linkerd viz uninstall [flags]

Examples:
linkerd viz uninstall | kubectl delete -f -

Flags:
  -h, --help   help for uninstall

Global Flags:
      --api-addr string            Override kubeconfig and communicate directly with the control plane at host:port (mostly for testing)
      --as string                  Username to impersonate for Kubernetes operations
      --as-group stringArray       Group to impersonate for Kubernetes operations
      --cni-namespace string       Namespace in which the Linkerd CNI plugin is installed (default "linkerd-cni")
      --context string             Name of the kubeconfig context to use
      --kubeconfig string          Path to the kubeconfig file to use for CLI requests
  -L, --linkerd-namespace string   Namespace in which Linkerd is installed (default "linkerd")
      --verbose                    Turn on debug logging

No resources found
```

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

**Review by kleimkuhler — COMMENTED** [review]
I'm back and forth on if it makes sense to print usage after the error, or just use `os.Exit` here so that usage is not printed.

The only case that it makes sense is if the user did not pass the `--context` flag when they should have. There are no other flags that they could pass that would change the fact that there are no resources to uninstall.

Did you consider this at all? I'm good with it as is, but curious if you had the same thoughts.

**Comment by adleong** [comment]
@kleimkuhler good point, this isn't really a usage error so it makes sense to omit the usage output.  I'll update.

**Review by kleimkuhler — APPROVED** [review]
Nice 👍

File: jaeger/cmd/uninstall.go [hunk 1]
```diff
@@ -34,17 +38,5 @@ func uninstallRunE(ctx context.Context) error {
...
-		return err
-	}
-
-	for _, r := range resources {
-		if err := r.RenderResource(os.Stdout); err != nil {
-			return fmt.Errorf("error rendering Kubernetes resource: %v", err)
-		}
-	}
-	return nil
+	return pkgCmd.Uninstall(ctx, k8sAPI, "linkerd.io/extension=jaeger")
```
Comment (dadjeibaah) [inline jaeger/cmd/uninstall.go:41]: We can copy the same format we have in viz's `uninstall` command here as well same with `multicluster`:
```suggestion
	return pkgCmd.Uninstall(ctx, k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, JaegerExtensionName))
```

Here is the full diff of the changes:
```diff
diff --git a/jaeger/cmd/uninstall.go b/jaeger/cmd/uninstall.go
index 32e119d8..f03592e0 100644
--- a/jaeger/cmd/uninstall.go
+++ b/jaeger/cmd/uninstall.go
@@ -38,5 +38,5 @@ func uninstallRunE(ctx context.Context) error {
                return err
        }

-       return pkgCmd.Uninstall(ctx, k8sAPI, "linkerd.io/extension=jaeger")
+       return pkgCmd.Uninstall(ctx, k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, JaegerExtensionName))
 }
diff --git a/multicluster/cmd/uninstall.go b/multicluster/cmd/uninstall.go
index 15e43529..1cef3131 100644
--- a/multicluster/cmd/uninstall.go
+++ b/multicluster/cmd/uninstall.go
@@ -53,7 +53,7 @@ func newMulticlusterUninstallCommand() *cobra.Command {
                                return errors.New(strings.Join(err, "\n"))
                        }

-                       err = pkgCmd.Uninstall(cmd.Context(), k8sAPI, "linkerd.io/extension=multicluster")
+                       err = pkgCmd.Uninstall(cmd.Context(), k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, MulticlusterExtensionName))
                        if err != nil {
                                fmt.Fprintln(os.Stderr, err)
                                os.Exit(1)
```

**Review by dadjeibaah — COMMENTED** [review]
This looks great and works well too! I left a few comments around formatting mostly

File: multicluster/cmd/uninstall.go [hunk 2]
```diff
@@ -59,28 +53,14 @@ func newMulticlusterUninstallCommand() *cobra.Command {
 				return errors.New(strings.Join(err, "\n"))
 			}
 
-			return uninstallRunE(cmd.Context(), k8sAPI)
+			err = pkgCmd.Uninstall(cmd.Context(), k8sAPI, "linkerd.io/extension=multicluster")
```
Comment (dadjeibaah) [inline multicluster/cmd/uninstall.go:56]:
```suggestion
			err = pkgCmd.Uninstall(cmd.Context(), k8sAPI, fmt.Sprintf("%s=%s", k8s.LinkerdExtensionLabel, MulticlusterExtensionName))
```

**Review by dadjeibaah — APPROVED** [review]
One more suggestion and this is good to go. Thanks for making those updates 📦 🚢

**Pull request merged**